src/main/java/org/terasology/las/platform/GuidanceNpcProvider.java

```diff
@@ -36,19 +37,16 @@
     @Override
     public void process(Region region, EntityBuffer buffer) {
         if (region.getRegion().encompasses(0, 64, 24)) {
-            Prefab dialog = Assets.getPrefab("LightAndShadow:TeamDialog").get();
-            DialogComponent dialogComp = dialog.getComponent(DialogComponent.class);
 
             Prefab chooseFactionNpc = Assets.getPrefab("LightAndShadow:MagicFool").get();
             EntityStore entity = new EntityStore(chooseFactionNpc);
             entity.addComponent(new LocationComponent(new Vector3f(0, 61, 24)));
-            entity.addComponent(dialogComp);
             buffer.enqueue(entity);
 
             Prefab beaconMark = Assets.getPrefab("BeaconMark").get();
             EntityStore entityStore = new EntityStore(beaconMark);
             entityStore.addComponent(new LocationComponent(new Vector3f(0, 63, 24)));
-
+            entityStore.addComponent(new NetworkComponent());
```

Comment:
Is there a reason for adding this component through code over prefab?

Reply:
'BeaconMark' prefab is taken from Quests module. Hence, adding network component would require changing that module. Should I create beaconmark prefab for this module too?

Reply:
I'd suggest looking into "delta"s. They allow you to modify a prefab from another module.
[See this guide for more info](https://github.com/Terasology/TutorialAssetSystem/wiki/Deltas-and-Overrides)